src/adapters/eddy/execution/src/index.ts

```diff
@@ -0,0 +1,115 @@
+const WETH = "0x8280a4e7D5B3B658ec4580d3Bc30f5e50454F169";
+const PUFF_ETH = "0x1B49eCf1A8323Db4abf48b2F5EFaA33F7DdAB3FC";
+const USDC = "0x1a1A3b2ff016332e866787B311fcB63928464509";
+const USDT = "0x2F8A25ac62179B31D62D7F80884AE57464699059";
+const WBTC = "0xDa4AaEd3A53962c83B35697Cd138cc6df43aF71f";
+const NATIVE_ETH = "0x000000000000000000000000000000000000800A";
+
+export const getUserTransactionData = async (startBlock: number, endBlock: number): Promise<any> => {
+    console.log(`Get Tx Data From ${startBlock} to ${endBlock}`);
+
+    const EddySwapRouter = "0x35F998694e553B772E0c22dC9CEb9bE0a1Bf8528";
+
+    const query = `query MyQuery {
+        eddySwaps(where: {
+          blockNumber_gte: ${startBlock},
+          blockNumber_lte: ${endBlock}
+        }){
+          id
+          amountIn
+          chainId
+          priceOfTokenIn
+          amountOut
+          blockNumber
+          blockTimestamp
+          tokenIn
+          tokenOut
+          walletAddress
+          fees
+          transactionHash
+        }
+      }`;
+
+    const response = await fetch('http://3.114.68.110:8000/subgraphs/name/eddyfinance-v01Fin', {
+            method: 'POST',
```

Comment:
minor issue, please fix it. Refer to the `fetchGraphQLData` function in the **example** and replace the endpoint here with the one from the .env environment variable. Also, add a _retry_ feature to handle data fetch failures due to network issue.

Reply:
Fixed